File: frontend/src/types/review.ts
// Multi-LLM Review Types

export interface LLMReview {
  id: string;
  llmProvider: 'claude' | 'codex' | 'gemini';
  score: number;
  reasoning: string;
  timestamp: string;
  metadata?: Record<string, unknown>;
}

export interface ReviewConsensus {
  averageScore: number;
  agreementLevel: 'high' | 'medium' | 'low';
  scores: number[];
  disagreements: string[];
}

export interface Appeal {
  id: string;
  submissionId: string;
  reviewerId: string;
  reason: string;
  status: 'pending' | 'under_review' | 'resolved' | 'rejected';
  createdAt: string;
  updatedAt: string;
  history: AppealHistory[];
}

export interface AppealHistory {
  id: string;
  action: string;
  actor: string;
  timestamp: string;
  notes?: string;
}

export interface ReviewDashboard {
  submissionId: string;
  reviews: LLMReview[];
  consensus: ReviewConsensus;
  appeal?: Appeal;
}

File: backend/models/review.py
"""Review system models."""

from pydantic import BaseModel, Field
from typing import List, Optional, Literal
from datetime import datetime


class LLMReview(BaseModel):
    id: str
    llm_provider: Literal['claude', 'codex', 'gemini']
    score: float = Field(ge=0, le=100)
    reasoning: str
    timestamp: datetime
    metadata: Optional[dict] = None


class ReviewConsensus(BaseModel):
    average_score: float
    agreement_level: Literal['high', 'medium', 'low']
    scores: List[float]
    disagreements: List[str]


class AppealHistory(BaseModel):
    id: str
    action: str
    actor: str
    timestamp: datetime
    notes: Optional[str] = None


class Appeal(BaseModel):
    id: str
    submission_id: str
    reviewer_id: Optional[str] = None
    reason: str
    status: Literal['pending', 'under_review', 'resolved', 'rejected']
    created_at: datetime
    updated_at: datetime
    history: List[AppealHistory]


class ReviewDashboard(BaseModel):
    submission_id: str
    reviews: List[LLMReview]
    consensus: ReviewConsensus
    appeal: Optional[Appeal] = None


class AppealCreate(BaseModel):
    submission_id: str
    reason: str


class AppealStatusUpdate(BaseModel):
    status: Literal['under_review', 'resolved', 'rejected']
    notes: Optional[str] = None


class ReviewerAssignment(BaseModel):
    reviewer_id: str
